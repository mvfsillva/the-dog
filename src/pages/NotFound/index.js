import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Title from '@components/Title'
import Button from '@components/Button'
import FullContainer from '~/styles/FullContainer'

import Bowl from './assets/Bowl'

const StyledTitle = styled(Title)`
  color: ${props => props.theme.colors.gray.default};
  font-family: ${props => props.theme.fonts.primary};
  font-weight: 300;
  line-height: 60px;
  letter-spacing: 3px;
`
const NotFound = ({ handleclick }) => (
  <FullContainer>
    <Bowl />
    <StyledTitle size="h2">Desculpe, mas essa página não existe.</StyledTitle>
    <Button onClick={e => (handleclick ? handleclick(e) : window.location.replace('/'))} outlined>
      Voltar para o inicio
    </Button>
  </FullContainer>
)

NotFound.propTypes = {
  handleclick: PropTypes.func,
}

NotFound.defaultProps = {
  handleclick: null,
}

export default NotFound
