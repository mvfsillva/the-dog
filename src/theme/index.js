const baselineGrid = 4
const interfaceGrid = 8

const breakpoints = ['576px', '768px', '992px', '1200px']

const colors = {
  primary: {
    default: '#6800c3',
    fallback: '#a55cff',
    text: '#fff',
  },
}

export default {
  baselineGrid,
  interfaceGrid,
  breakpoints,
  colors,
}
