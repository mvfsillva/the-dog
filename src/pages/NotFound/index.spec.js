import React from 'react'
import 'jest-styled-components'
import theme from 'src/theme'
import { shallowWithTheme } from '../../../config/test/enzyme.theme'
import NotFound from '.'

const shallow = shallowWithTheme(theme)

const mockAction = jest.fn()
const mockReplace = jest.fn()

const wrap = shallow(<NotFound />)

describe('[Page: NotFound]', () => {
  it('executes replace if handleclick not present', () => {
    global.location.replace = mockReplace
    wrap.find('Button').simulate('click')
    expect(mockReplace).toHaveBeenCalled()
    expect(mockAction).not.toHaveBeenCalled()
  })

  it('execute click action', () => {
    wrap.setProps({ handleclick: mockAction })
    wrap.find('Button').simulate('click')
    expect(mockAction).toHaveBeenCalled()
    expect(mockReplace).not.toHaveBeenCalled()
  })

  afterEach(() => {
    jest.clearAllMocks()
  })
})
