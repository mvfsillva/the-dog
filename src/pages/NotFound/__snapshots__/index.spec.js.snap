// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Page: NotFound] matches snapshot 1`] = `
<styled.div>
  <Bowl />
  <Styled(Title)
    size="h2"
  >
    Desculpe, mas essa página não existe.
  </Styled(Title)>
  <Button
    height={40}
    onClick={[Function]}
    outlined={true}
    type="button"
  >
    Voltar para o inicio
  </Button>
</styled.div>
`;
